src/fhirtypes/common.ts

```diff
@@ -205,6 +213,41 @@ export function replaceField(
   }
 }
 
+/**
+ * Adds Mixin rules onto a Profile, Extension, or Instance
+ * @param {Profile | Extension | Instance} fshDefinition - The definition to apply mixin rules on
+ * @param {FSHTank} tank - The FSHTank containing the fshDefinition
+ */
+export function applyMixinRules(
+  fshDefinition: Profile | Extension | Instance,
+  tank: FSHTank
+): void {
+  // Rules are added to beginning of rules array, so add the last mixin rules first
+  [...fshDefinition.mixins].reverse().forEach(mixinName => {
+    const ruleSet = tank.fish(mixinName, Type.RuleSet) as RuleSet;
+    if (ruleSet) {
+      ruleSet.rules.forEach(r => {
+        // Record source information of Profile/Extension/Instance on which Mixin is applied
+        r.sourceInfo.appliedFile = fshDefinition.sourceInfo.file;
+        r.sourceInfo.appliedLocation = fshDefinition.sourceInfo.location;
+      });
+      const rules = ruleSet.rules.filter(r => {
+        if (fshDefinition instanceof Instance && !(r instanceof FixedValueRule)) {
+          logger.error(
+            'Mixin rules applied to an instance must fix a value. Other rules are ignored.',
+            r.sourceInfo
+          );
+          return false;
+        }
+        return true;
+      });
+      fshDefinition.rules = [...rules, ...fshDefinition.rules];
+    } else {
+      logger.error(`Unable to find definition for Mixin ${mixinName}.`, fshDefinition.sourceInfo);
```

Comment:
Until we support other things, should this be: _Unable to find definition for RuleSet ${mixinName}_?

Reply:
Yeah that makes sense, I'll change it.